src/blockdata/script.rs

```diff
@@ -75,6 +75,22 @@ impl fmt::UpperHex for Script {
     }
 }
 
+impl hex::FromHex for Script {
+    fn from_byte_iter<I>(iter: I) -> Result<Self, hex::Error>
+        where I: Iterator<Item=Result<u8, hex::Error>> +
+            ExactSizeIterator +
+            DoubleEndedIterator,
+    {
+        Ok(Script(Vec::from_byte_iter(iter)?.into()))
```

Comment:
if you wish to skip the ? unrwapping... Vec::from_byte_iter(iter).map(Box<[u8]>::from).map(Script)

Reply:
Nice. Yeah, I'll change to do that.